File: cpp/54d_1846A_RudolphAndCutTheRope.cpp
#include<bits/stdc++.h>

int solve(int n){
    int a, b, cnt = 0;
    std::vector<int> hstack;
    for(int i = 0; i < n; i++){
        std::cin>>a>>b;
        hstack.push_back(a - b);
    }
    sort(hstack.begin(), hstack.end());
    while(hstack.back() > 0){
        cnt ++;
        hstack.pop_back();
    }
    return cnt;
}

int main(){
    int t, n;
    std::vector<int> ans;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::cin>>n;
        ans.push_back(solve(n));
    }
    for(auto i : ans){
        std::cout<<i<<'\n';
    }
    return 0;
}

File: python/54d_1846A_RudolphAndCutTheRope.py
def solve(n):
    cnt = 0
    hstack = []
    for _ in range(n):
        a, b = input().split()
        a, b = int(a), int(b)
        hstack.append(a - b)
    hstack.sort()
    while len(hstack) > 0 and hstack[-1] > 0:
        cnt += 1
        hstack.pop()
    return cnt

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        ans.append(solve(n))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()